staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -194,10 +210,59 @@ func (s *SpdyRoundTripper) dial(req *http.Request) (net.Conn, error) {
 
 	rwc, _ := proxyClientConn.Hijack()
 
-	if req.URL.Scheme != "https" {
-		return rwc, nil
+	if req.URL.Scheme == "https" {
+		return s.tlsConn(rwc, targetHost)
```

Comment:
plumb `proxyReq.Context()` to `tlsConn` so we can pass it to `tlsConn.HandshakeContext()`

Reply:
done.